File: JUnitTestingLastAct/src/Decorator/DecoratorTestRunner.java
package Decorator;


import Singleton.SingletonTestCase;
import org.testng.annotations.Test;

import java.util.logging.Logger;

import static org.testng.AssertJUnit.assertTrue;

public class DecoratorTestRunner {

    private final Logger logger = Logger.getLogger(DecoratorTestRunner.class.getName());


    @Test
    public final void rectangleIsShape() {
        ShapeDecorator newRectangle = new ShapeDecorator(new Rectangle()) {
        };
        assertTrue(newRectangle instanceof Shape);
        this.logger.info("Rectangle is an instance of Shape");
    }

    @Test
    public final void circleIsShape(){
        ShapeDecorator newCircle = new ShapeDecorator(new Circle()){
        };
        assertTrue(newCircle instanceof Shape);
        this.logger.info("Circle is an instance of Shape");
    }

    @Test
    public final void decoratorIsShape() {
        ShapeDecorator newShapeDecorator = new ShapeDecorator(new Rectangle()) {
        };
        assertTrue(newShapeDecorator instanceof Shape);
        this.logger.info("newShapeDecorator is an instance of Shape");
    }

    @Test
    public final void redShapeDecoratorIsShape() {
        RedShapeDecorator newRedShapeDecorator = new RedShapeDecorator(new Rectangle()) {
        };
        assertTrue(newRedShapeDecorator instanceof Shape);
        this.logger.info("redShapeDecorator is an instance of Shape");
    }


}

File: KotlinTesting/src/test/structural/DecoratorTestRunner.kt
package Decorator

import org.junit.Test

import java.util.logging.Logger
import kotlin.test.assertTrue

class DecoratorTestRunner {
    private val logger = Logger.getLogger(DecoratorTestRunner::class.java.name)
    @Test
    fun rectangleIsShape() {
        val newRectangle: ShapesDecorator = object : ShapesDecorator(Rectangle()) {}
        assertTrue(newRectangle is Shape)
        logger.info("Rectangle is an instance of Shape")
    }

    @Test
    fun circleIsShape() {
        val newCircle = object : ShapesDecorator(Circle()) {}
        assertTrue(newCircle is Shape)
        logger.info("Circle is an instance of Shape")
    }

    @Test
    fun decoratorIsShape() {
        val newShapeDecorator = object : ShapesDecorator(Rectangle()) {}
        assertTrue(newShapeDecorator is Shape)
        logger.info("newShapeDecorator is an instance of Shape")
    }

    @Test
    fun redShapeDecoratorIsShape() {
        val newRedShapeDecorator = object : RedShapeDecorator(Rectangle()) {}
        assertTrue(newRedShapeDecorator is Shape)
        logger.info("redShapeDecorator is an instance of Shape")
    }
}